research-software-directory/RSD-as-a-service
# chore: upgrade eslint to v9

# Upgrade eslint to v9

Closes #1198 

Changes proposed in this pull request:
* upgrade eslint to v9.13.0
* migrate configuration to eslint.config.mjs
* disable rules that cause new linter warnings/errors. 
* Note: we will enable some of these rules later and fix linting errors

**I would propose to just update eslint and test if everything works properly. I would prefer to handle prettier in a separate PR.**

How to test:
* This should be **tested from the frontend folder**. You need node v20+ in order to perfom these tests.
* If you have old node modules remove these: `rm -rf node_modules`
* Install dependencies `npm install`
* run linter `npm run lint`
* No errors should be found. 
* Now add console.log in any of the frontend files and run linter again
* It should warn for console logs. 
* run tests `npm test`. All test should pass
* run build `npm run build`. Build should work.
* run complete RSD setup, `make start`, RSD should work without problems.

PR Checklist:
